packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -789,3 +806,56 @@ export async function DocsQA<ChunkMetadata extends Jsonifiable = Jsonifiable>(pr
     </ChatCompletion>
   );
 }
+
+const ResultSchema: z.ZodObject<any> = z
```

Comment:
```suggestion
const ResultSchema = z
```

I know this was copied from somewhere else where it was needed, but in this case, we don't need it. (It was only needed when we were doing a recursive type.)

The type annotation is more generic than reality, so our types end up sadly looser:

![image](https://github.com/fixie-ai/ai-jsx/assets/829827/be05ea06-402e-4483-a490-ac35b1942d12)
![image](https://github.com/fixie-ai/ai-jsx/assets/829827/7d0b3520-b93b-4461-92f5-ed9e9aa6d9fa)

When we omit the type annotation, the types flow through nicely:

![image](https://github.com/fixie-ai/ai-jsx/assets/829827/22503fb0-3f20-4495-9fab-8870aef2b926)
![image](https://github.com/fixie-ai/ai-jsx/assets/829827/375bb118-92ec-4fc6-9351-9465a539f060)

And of course these nicer types flow through to userland.